src/main/java/it/smartcommunitylab/aac/openidfed/provider/OpenIdFedIdentityProvider.java

```diff
@@ -173,13 +176,30 @@ protected OIDCUserIdentity buildIdentity(
 
     @Override
     public String getAuthenticationUrl() {
-        return "/auth/" + getAuthority() + "/authorize/" + getProvider();
+        return getLoginUrl();
+    }
+
+    @Override
+    public String getAuthenticationUrl(HttpServletRequest request) {
```

Comment:
the parameter should be plaintext. The logic for addressing the composite identifier is an internal one, we don't need to expose it to clients.

A request should look like
```
...authorize?idp_hint=PROVIDER_ID&domain_hint=https://spid.test
```

and then inside the method we can apply the encoding required to build the redirect.